### PR TITLE
ENH JTFS `frequency_averaging`

### PR DESCRIPTION
**Pull request by lostanlen**
#945's frequential sibling

while writing this i found a metadata bug in `frequency_scattering` (#935), i.e. that we were returning `j_fr` and `n_fr` as integers instead of 1-tuples. This was unnecessarily inconsistent with first-order time scattering. So i fixed that and updated the tests. 

This is the last element of core JTFS. When #945 and this PR will be merged, i'll be able to work on a `scattering` method for the base frontend.

### CODE REVIEW

File: kymatio/scattering1d/core/timefrequency_scattering.py [hunk 1]
```diff
@@ -275,4 +275,63 @@ def frequency_scattering(X, backend, filters_fr, oversampling_fr,
         # as (X['n'][1:] + (n_fr,)), i.e., n=(n_fr,) is not spinned
         # and n=(n2, n_fr) if spinned. This 'n' tuple is unique.
         yield {**X, 'coef': Y_fr, 'n': (X['n'][1:] + (n_fr,)),
-            'j_fr': j_fr, 'n_fr': n_fr, 'spin': spin}
+            'j_fr': (j_fr,), 'n_fr': (n_fr,), 'spin': spin}
+
+
+def frequency_averaging(U_2, backend, phi_fr_f, oversampling_fr, average_fr):
```
Comment (changhongw) [inline kymatio/scattering1d/core/timefrequency_scattering.py:281]: like `phi_fr_f` :)

File: tests/scattering1d/test_timefrequency_scattering.py [hunk 2]
```diff
@@ -156,7 +157,7 @@ def test_frequency_scattering():
         jtfs.oversampling_fr, jtfs.average_fr=='local', spinned=True)
     for Y_fr in freq_gen:
         assert Y_fr['coef'].shape[-1] == X['coef'].shape[-1]
-        assert Y_fr['n'] == (4, Y_fr['n_fr'])
+        assert Y_fr['n'] == ((4,) + Y_fr['n_fr'])
```
Comment (changhongw) [inline tests/scattering1d/test_timefrequency_scattering.py:160]: redundant `()` on the right => `assert Y_fr['n'] == (4,) + Y_fr['n_fr']`

**Review by changhongw — APPROVED** [review]

File: tests/scattering1d/test_timefrequency_scattering.py (hunk 2 above)
Comment (lostanlen) [reply tests/scattering1d/test_timefrequency_scattering.py:160]: it's OK

Comment (lostanlen) [reply tests/scattering1d/test_timefrequency_scattering.py]: 5813718

**Review by cyrusvahidi — APPROVED** [review]

**Comment by lostanlen** [comment]
... and this concludes core JTFS 🥳 thanks everyone

**Pull request merged**